glyphs-reader/src/plist.rs

```diff
@@ -351,8 +369,43 @@ impl Plist {
             Plist::Float(f) => {
                 s.push_str(&format!("{f}"));
             }
+            Plist::Data(data) => {
+                s.push('<');
+                for byte in data {
+                    s.extend(hex_digits_for_byte(*byte))
+                }
+            }
+        }
+    }
+}
+
+fn hex_digits_for_byte(byte: u8) -> [char; 2] {
+    fn to_hex_digit(val: u8) -> char {
+        match val {
+            0..=9 => ('0' as u32 as u8 + val).into(),
+            10..=15 => (('a' as u32 as u8) + val - 10).into(),
+            _ => unreachable!("only called with values in range 0..=15"),
+        }
+    }
+
+    [to_hex_digit(byte >> 4), to_hex_digit(byte & 0x0f)]
+}
+
+fn byte_from_hex(hex: [u8; 2]) -> Result<u8, Error> {
```

Comment:
I feel like I've written something like this often enough that it should be in core.